### PR TITLE
CAMEL-10538 Add declarative validator according to input/output type

### PR DESCRIPTION
**Pull request by igarashitm**

### CODE REVIEW

**Comment by igarashitm** [comment]
Adding a last piece of type awareness work at this point, declarative content validation. I hope this document explains well.
https://github.com/igarashitm/camel/blob/45cb9c226084a66b15bb980671fe756cac424f16/camel-core/src/main/docs/validator.adoc

File: camel-core/src/main/docs/validator.adoc [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+-------------------------------------------------------------------
+validator()
+    .type("csv:CSVOrder")
+    .withExpression(bodyAs(String.class).contains("{name:XOrder}"));
+-------------------------------------------------------------------
+
+XML DSL:
+[source,xml]
+-------------------------------------------------------------------
+<predicateValidator Type="csv:CSVOrder">
```
Comment (davsclaus) [inline camel-core/src/main/docs/validator.adoc]: These xml tags dont match

File: camel-core/src/main/docs/validator.adoc [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+| ref | Reference to the link:endpoint.html[Endpoint] ID
+
+| uri | link:endpoint.html[Endpoint] URI
+|=======================================================================
+Here is an example to specify endpoint URI in Java DSL:
+[source,java]
+-------------------------------------------------------------------
+validator()
+    .type("xml")
+    .withUri("validator:xsd/schema.xsd");
```
Comment (davsclaus) [inline camel-core/src/main/docs/validator.adoc]: In this example we may want to highlight that the `validator` endpoint being used is the Camel validator component that has been with Camel for a very long time. And that you can use any other Camel endpoint if you want to such as bean-valdator or any custom component

File: camel-core/src/main/docs/validator.adoc [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+-------------------------------------------------------------------
+
+If you have following route definition, above validator will be applied when `direct:abc` endpoint
+receives the message. Note that the `validate` attribute on the inputType declaration is set to `true`:
+
+Java DSL:
+[source,java]
+-------------------------------------------------------------------
+from("direct:abc")
+    .inputType("xml:ABCOrder", true)
```
Comment (davsclaus) [inline camel-core/src/main/docs/validator.adoc]: Should we have `inputTypeWithValidate` as well to make it more clear validate is in use, than a boolean true parameter would do?

Comment (igarashitm) [inline camel-core/src/main/docs/validator.adoc]: Thanks for the good catches Claus! especially the `inputTypeWithValidate` is much more descriptive in Java DSL. I'll fix those soon.

**Comment by igarashitm** [comment]
OK fixed - I was a bit wondering if I should introduce `<inputTypeWithValidate>` for XML DSL as well to align with Java DSL, but I believe there's no problem as the attribute `validate="true"` on inputType/outputType looks fine unlike the boolean parameter in Java DSL.

**Comment by davsclaus** [comment]
Yeah the XML DSL is fine. XML attributes is better there, and also how we have done this with other stuff in Camel

**Comment by davsclaus** [comment]
Thanks for the PR it has been merged. Do you mind closing this?

**Comment by igarashitm** [comment]
Thanks! closing.

**Pull request closed without merging**